src/include/migraphx/op/concat.hpp

```diff
@@ -73,49 +74,96 @@ struct concat
         }
         return offsets;
     }
+
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
-        if(inputs.empty())
+        // inputs can contain 1 or more shapes (variadic).  compute_shape_op ensures there must
+        // be at least 1.
+
+        if(not std::all_of(inputs.begin(), inputs.end(), [&](const shape& s) {
+               return (s.ndim() == inputs[0].ndim() and s.type() == inputs[0].type());
+           }))
```

Comment:
This should be checked with `check_shapes{inputs, *this}.same_ndim().same_type()`.

Reply:
changed